src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -517,6 +517,20 @@ impl Visitor<'tcx> for Validator<'_, 'mir, 'tcx> {
                     return;
                 }
 
+                // See if this is a trait method for a concrete type whose impl of that trait is
+                // `const`.
+                if self.tcx.features().const_trait_impl {
+                    let instance = Instance::resolve(self.tcx, self.param_env, def_id, substs);
```

Comment:
As long as `self.param_env` is `tcx.param_env(some_def_id)` *without* calling `.with_reveal_all()` or w/e, it should be fine.